### PR TITLE
Added solution to HW6-2

### PR DESCRIPTION
**Pull request by MichaelShevchenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,26 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other != null && getClass() == other.getClass() && this == other) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: these should be separate checks
they make no sense if put all together

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,26 @@ public Integer getLength() {
...
+
+        Rectangle rect = (Rectangle) other;
+        return (width == rect.width || (width != null && width.equals(rect.getWidth())))
+                && (length == rect.length || (length != null && length.equals(rect.getLength())))
+                && (color == rect.color || (color != null && color.equals(rect.getColor())));
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 42;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: is it a prime number?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (MichaelShevchenko) [reply src/main/java/core/basesyntax/Rectangle.java]: done

File: src/main/java/core/basesyntax/Rectangle.java (hunk 2 above)
Comment (MichaelShevchenko) [reply src/main/java/core/basesyntax/Rectangle.java]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**